### ISSUE TITLE
Let's talk about CI/CD & E2E tests

### ISSUE THREAD

**Issue by ROpdebee** [ROpdebee/mb-userscripts#97]
I've rushed the implementation of CD quite a bit, mainly because I wanted to get it done and it was already getting late. In retrospect, the setup isn't the greatest, so let's talk about it.

Currently, these are the workflows that are running on this repo:
- `main`: runs on each PR and does the following jobs:
    - lint & typecheck (in parallel)
    - build
    - run unit tests
- `nightly tests`: runs every night on the `main` branch, same jobs as before, but also:
    - create an issue if any job failed, to track CI failures
- `deploy`: runs after a PR is merged, does the same jobs as `main`, as well as:
    - if all jobs succeed, deploy new userscript versions if necessary
    - report deployment status in the merged PR

There's a couple of problems though:
- CI jobs are a bit slow because we're running most jobs sequentially while they could possibly be run in parallel
- We don't have an action that runs on every commit to the `main` branch, so commits on `main` don't get a CI checkmark until nightly tests are run, and coverage data on codecov is outdated until the nightly tests.
- When we add E2E tests, where will those fit into the picture?

Here's what I propose:
- Run lint, typecheck, build and unit test all in parallel, since there are no dependencies between them.
- Run the `main` workflow on pushes to `main` again, as we did before.
- `deploy` remains the same (technically it shouldn't be necessary to test anymore, as those are already done in the PR, but it's a good idea to keep them just to be sure).
- Add a `weekly tests` workflow that does the HTTP passthrough tests so we can periodically check when provider responses change etc.

For the E2E tests, we can split those up into two parts, as I proposed in #20:
1. Thorough tests on a couple of popular browser/userscript combos
2. Superficial tests on other browsers

We'll then select a few of the browser/engine combos of (1) to run in the `main` workflow (both on pushes to the `main` branch and each PR commit). The rest of the browser/engine combos, and the tests of (2) could then be run in `deploy` (if anything fails, it'll be commented in the merged PR) and in the `weekly tests`, again to discover any upstream changes.

I think that makes a bit more sense than the workflows we're running currently, but I'll have to think about it a bit more. I just don't want to forget what I already thought about :)
